llvm/include/llvm/SYCLLowerIR/ESIMD/Allocator.h

```diff
@@ -0,0 +1,53 @@
+//===--------- Allocator.h - Allocator for demangler
```

Comment:
I should've expressed my proposal clearer, sorry. This is just implementation detail of demangling code, not a general purpose service, so I would very much prefer this to reside in `ESIMDUtils.cpp`/`ESIMDUtils.hpp`